students/luke/s04/file_exercise.py

```diff
@@ -0,0 +1,25 @@
+#!/usr/bin/env python3
+
+"""
+https://uwpce-pythoncert.github.io/PythonCertDevel/exercises/file_lab.html#exercise-file-lab
+
+Write a little script that reads that file, and generates a list of all the languages that have been used.
+"""
+
+infile = open("students.txt", 'r')
+
+languages = { }
+
+for line in infile:
+    # split at :
+    name, sep, langs = line.partition(':')
```

Comment:
What is the benefit of using partition instead of split, if you're not using the "sep" variable?

Reply:
partition was listed first at https://docs.python.org/3/library/stdtypes.html#string-methods and its use was more immediately obvious than split.  It does look like split is just the thing, though.  Thanks, fixed in next revision.

Reply:
Split is nice when you don't need back the delimiter, but partition does the job here, as well :)